protected/humhub/modules/stream/models/filters/ModuleStreamFilter.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\stream\models\filters;
+
+use humhub\models\Setting;
+use humhub\modules\content\components\ContentContainerActiveRecord;
+use humhub\modules\content\components\ContentContainerModule;
+use humhub\modules\content\models\ContentContainerModuleState;
+use humhub\modules\space\models\Space;
+use humhub\modules\user\models\User;
+use ReflectionClass;
+use Yii;
+
+class ModuleStreamFilter extends StreamQueryFilter
+{
+    public function apply(): void
+    {
+        if ($this->streamQuery->container instanceof ContentContainerActiveRecord) {
+            $containerClass = $this->streamQuery->container instanceof Space ? Space::class : User::class;
+            $this->excludeContainersWithNotAvailableModule($containerClass);
+        }
+    }
+
+    private function excludeContainersWithNotAvailableModule(string $containerClass): void
+    {
+        $reflectClass = new ReflectionClass($containerClass);
+
+        // Find modules with "Not Available" option per User/Space
+        $moduleIds = Setting::find()
+            ->select('module_id')
+            ->where(['name' => 'moduleManager.defaultState.' . $reflectClass->getShortName()])
+            ->andWhere(['value' => ContentContainerModuleState::STATE_NOT_AVAILABLE])
+            ->column();
+
+        if (empty($moduleIds)) {
+            return;
+        }
+
+        $excludedContentClasses = [];
+        foreach ($moduleIds as $moduleId) {
+            $module = Yii::$app->getModule($moduleId);
+            if ($module instanceof ContentContainerModule) {
+                $excludedContentClasses = array_merge($excludedContentClasses, $module->getContentClasses());
```

Comment:
@funkycram @luke- I think we should be sure that all modules extended from `ContentContainerModule` properly define the classes in the method `getContentClasses()`, because during tests I find at least one module is "Files" https://github.com/humhub/cfiles/blob/master/Module.php which doesn't there records like `humhub\modules\cfiles\models\File::class`.

I didn't investigate deeply why it was missed there, probably some reason exists.
But if we set the module "Files" to new option "Not available" then the `File` records will not be hidden from a wall stream as expected, because the records will not be excluded because of `humhub\modules\cfiles\Module->getContentClasses()` returns empty array now.

Reply:
Discussed here https://github.com/humhub/humhub/issues/6486#issuecomment-1678532487